SmartDeviceLinkTests/RPCSpecs/PayloadSpecs/SDLSecurityQueryPayloadSpec.m

```diff
@@ -0,0 +1,80 @@
+//
+//  SDLSecurityQueryPayloadSpec.m
+//  SmartDeviceLinkTests
+//
+//  Created by Frank Elias on 8/12/21.
+//  Copyright © 2021 smartdevicelink. All rights reserved.
+//
+
+
+#import <Foundation/Foundation.h>
+
+#import <Quick/Quick.h>
+#import <Nimble/Nimble.h>
+
+#import "SDLRPCParameterNames.h"
+#import "SDLRPCFunctionNames.h"
+#import "SDLSecurityQueryPayload.h"
+#import "SDLSecurityQueryErrorCode.h"
+
+QuickSpecBegin(SDLSecurityQueryPayloadSpec)
+
+__block SDLSecurityQueryPayload* testPayload;
+__block NSDictionary* dict = @{@"id": @"3", @"text": @"SDL does not support encryption"};
+
+NSData* (^testData)(void) = ^NSData* {
+    NSData* jsonData = [NSJSONSerialization dataWithJSONObject:dict options:0 error:0];
+    NSData* binaryData = [NSData dataWithBytes:"PrimitiveString" length:strlen("PrimitiveString")];
+
+    Byte header[12] = {0x20, 0x00, 0x00, 0x02, 0x00, 0x00, 0x14, 0x43, 0x00, 0x00, 0x00, 0x00};
+    *(UInt32 *)&header[8] = CFSwapInt32HostToBig((unsigned int)jsonData.length);
+
+    NSMutableData *data = [NSMutableData dataWithCapacity:12 + jsonData.length];
```

Comment:
If you want to allocate data buffer beforehand then you had better do it a bit differently:
```
NSMutableData *data = [NSMutableData dataWithCapacity:sizeof(header) + jsonData.length + binaryData.length];
```